core/src/main/java/hudson/model/Run.java

```diff
@@ -1394,11 +1394,13 @@ public String getLength() {
         }
         
         public long getFileSize(){
-            if (length.length() == 0) {
-                // case for symlink
+            try {
+                return Long.decode(length);
+            }
+            catch (NumberFormatException e) {
+                LOGGER.log(FINE, "The length {0} is a valid long.", length);
```

Comment:
Is *NOT* a valid long.
It also makes sense to add the path or something so that the Artifact can be determined from the log entry

Reply:
```suggestion
                LOGGER.log(FINE, "Cannot determine file size of the artifact {0}. The length {1} is not a valid long value", new Object[] {this, length});
```